src/apps/sommelier/positions.ts

```diff
@@ -0,0 +1,101 @@
+import { logger } from '../../log'
+import { getClient } from '../../runtime/client'
+import { getTokenId } from '../../runtime/getTokenId'
+import { toDecimalNumber } from '../../types/numbers'
+import {
+  PositionDefinition,
+  PositionsHook,
+  PricePerShareContext,
+  UnknownAppTokenError,
+} from '../../types/positions'
+import { cellarV0821Abi } from './abis/cellar'
+import { getSommStrategiesData } from './sommelierApi'
+
+const hook: PositionsHook = {
+  getInfo() {
+    return {
+      id: 'sommelier',
+      name: 'Sommelier',
+      description: 'Sommelier vaults',
+    }
+  },
+  async getPositionDefinitions({ networkId }) {
+    const client = getClient(networkId)
+    const cellars = await getSommStrategiesData(networkId)
+    const positionDefinitions: PositionDefinition[] = []
+
+    const results = await Promise.allSettled(
+      cellars.map(async (cellar) => {
+        const [underlyingAsset, underlyingAssetSymbol, cellarName] =
+          await Promise.all([
+            client.readContract({
+              address: cellar.address,
+              abi: cellarV0821Abi,
```

Comment:
I noticed the config on their website uses different ABIs based on the vault.
Do they still implement the 3 methods called? 
Or would we need to tweak it per vault?

Reply:
yes i added a comment about this so that we remember, there are indeed 4 different cellar abi's but they all implement the methods that we need here. the main differences are in the events they emit and the non read-only methods, which we don't care much about here so i thought it was not worth also keeping a mapping of abis for our purposes